my_enumerable.rb

```diff
@@ -0,0 +1,154 @@
+# rubocop: disable Metrics/ModuleLength:
+# rubocop: disable Style/For:
+# rubocop: disable Style/CaseEquality:
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    item = 0
+    array = to_a
+    n = array.length
+    while item < n
+      yield(array[item])
+      item += 1
+    end
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    i = 0
+    for item in self do
+      yield(item, i)
+      i += 1
+    end
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    new_arr = []
+    for item in self do
+      new_arr.push(item) if yield(item)
+    end
+    new_arr
+  end
+
+  def my_all?(args = nil)
+    if args.nil?
+      my_each do |item|
+        return false unless item
+      end
+    elsif block_given?
+      my_each do |item|
+        return false unless yield(item)
+      end
+    else
```

Comment:
As far as I can see the issue here is that you're missing the comparison needed to make it through...  
For example: 
return false unless item == 5. 
instead, you're just yielding the item. Hope this makes it clear now ;)

> Note: don't forget that in this method as my previous colleague mentioned it; it should return true if the block never returns false or nil 

###### Good Luck